ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/InMemoryKVParticipantStateIT.scala

```diff
@@ -130,88 +127,80 @@ class InMemoryKVParticipantStateIT
     }
 
     "provide update after allocateParty" in {
-      val rt = ps.getNewRecordTime()
-
-      val hint = Some("Alice")
+      val hint = Some(Ref.Party.assertFromString("Alice"))
       val displayName = Some("Alice Cooper")
 
       for {
-        allocResult <- ps.allocateParty(hint, displayName).toScala
+        allocResult <- ps.allocateParty(hint, displayName, randomLedgerString()).toScala
         updateTuple <- ps.stateUpdates(beginAfter = None).runWith(Sink.head)
       } yield {
-        allocResult match {
-          case PartyAllocationResult.Ok(partyDetails) =>
-            assert(partyDetails.party == hint.get)
-            assert(partyDetails.displayName == displayName)
-            assert(partyDetails.isLocal)
-          case _ =>
-            fail("unexpected response to party allocation")
-        }
+        assert(
+          allocResult == SubmissionResult.Acknowledged,
+          s"unexpected response to party allocation: $allocResult")
         updateTuple match {
           case (offset: Offset, update: PartyAddedToParticipant) =>
             assert(offset == Offset(Array(0L, 0L)))
             assert(update.party == hint.get)
             assert(update.displayName == displayName.get)
             assert(update.participantId == ps.participantId)
             assert(update.recordTime >= rt)
-          case _ => fail("unexpected update message after a party allocation")
+          case _ =>
+            fail(
+              s"unexpected update message after a party allocation. Error : ${allocResult.description}")
         }
       }
     }
 
     "accept allocateParty when hint is empty" in {
-      val hint = None
+      val hintNone = None
       val displayName = Some("Alice Cooper")
 
       for {
-        result <- ps.allocateParty(hint, displayName).toScala
+        result <- ps.allocateParty(hint = None, displayName, randomLedgerString()).toScala
+        updateTuple <- ps.stateUpdates(beginAfter = None).runWith(Sink.head)
       } yield {
-        result match {
-          case PartyAllocationResult.Ok(_) =>
-            succeed
+        assert(result == SubmissionResult.Acknowledged, "unexpected response to party allocation")
+        updateTuple match {
+          case (offset: Offset, update: PartyAddedToParticipant) =>
+            assert(offset == Offset(Array(0L, 0L)))
+            assert(update.party != hintNone)
```

Comment:
perhaps just use `assert(update.party.isSome)` or something here.

Reply:
Addressed in 165899128.